src/main/java/me/dinowernli/grpc/polyglot/command/ServiceList.java

```diff
@@ -20,97 +22,146 @@
 
   /** Lists the GRPC services - filtered by service name (contains) or method name (contains) */
   public static void listServices(
-      Output output,
-      FileDescriptorSet fileDescriptorSet,
-      String protoDiscoveryRoot,
-      Optional<String> serviceFilter,
-      Optional<String> methodFilter,
-      Optional<Boolean> withMessage) {
+    Output output,
+    FileDescriptorSet fileDescriptorSet,
+    String protoDiscoveryRoot,
+    Optional<String> serviceFilter,
+    Optional<String> methodFilter,
+    Optional<Boolean> withMessage,
+    Optional<Boolean> jsonOutput) {
+
+    ArrayList<Service> services = new ArrayList<>(
+      getServices(fileDescriptorSet, protoDiscoveryRoot, serviceFilter, methodFilter, withMessage, jsonOutput));
+
+    if (jsonOutput.isPresent() && jsonOutput.get()) {
+      Gson gson = new Gson();
+      output.writeLine(gson.toJson(services));
```

Comment:
As discussed above, can we not just json-print the descriptors rather than converting them to our custom data type?